system/jlib/jevent.cpp

```diff
@@ -1754,6 +1758,913 @@ IEventVisitor* createDumpCSVEventVisitor(IBufferedSerialOutputStream& out)
+        }
+    }
+
+protected:
+    StringAttr accepted;
+    std::regex pattern;
+    bool checkRegex{false};
+    EventType currentEvent{EventNone};
+    __uint64 currentIndex{0};
+    std::set<__uint64> acceptableIds;
+};
+
+// Filter term that rejects all events for attribute values other than the one given. A visited
+// attribute value must either match exactly or be a regular expression match with the term's
+// accepted value.
+class CTextAttributeFilterTerm : public CEventVisitFilterTerm
+{
+public:
+    virtual Continuation visitAttribute(EventAttr id, const char* value) override
+    {
+        if (id != attribute)
+            return visitContinue;
+        if (streq(accepted, value))
+            return visitContinue;
+        if (checkRegex && std::regex_match(value, pattern))
+            return visitContinue;
+        return visitSkipEvent;
+    }
+
+public:
+    CTextAttributeFilterTerm(EventAttr _attribute, const char* value)
+        : attribute(_attribute), accepted(value ? value : "")
+    {
+        try
+        {
+            pattern.assign(accepted.str());
+            checkRegex = true;
+        }
+        catch(const std::exception& e)
+        {
+            // ignore... the accepted value is not required to be a regular expression
+        }
+    }
+
+protected:
+    EventAttr attribute;
+    StringAttr accepted;
+    std::regex pattern;
+    bool checkRegex{false};
+};
+
+// Filter term that rejects all events for attribute values other than those given. Either a
+// single accepted value or a range of accepted values may be specified
+class CUnsignedAttributeFilterTerm : public CEventVisitFilterTerm
+{
+public:
+    virtual Continuation visitAttribute(EventAttr id, __uint64 value) override
+    {
+        if (id == attribute && (value < start || value > end))
+            return visitSkipEvent;
+        return visitContinue;
+    }
+
+public:
+    CUnsignedAttributeFilterTerm(EventAttr _attribute, __uint64 value)
+        : attribute(_attribute), start(value), end(value)
+    {
+    }
+
+    CUnsignedAttributeFilterTerm(EventAttr _attribute, __uint64 _start, __uint64 _end)
+        : attribute(_attribute), start(_start), end(_end)
+    {
+    }
+
+protected:
+    EventAttr attribute;
+    __uint64 start;
+    __uint64 end;
+};
+
+// Filter term that rejects all events for attribute values other than the given value
+class CBooleanAttributeFilterTerm : public CEventVisitFilterTerm
+{
+public:
+    virtual Continuation visitAttribute(EventAttr id, bool value) override
+    {
+        if (id == attribute && value != accepted)
+            return visitSkipEvent;
+        return visitContinue;
+    }
+
+public:
+    CBooleanAttributeFilterTerm(EventAttr _attribute, bool _accepted)
+        : attribute(_attribute), accepted(_accepted)
+    {
+    }
+
+protected:
+    EventAttr attribute;
+    bool accepted;
+};
+
+class CLogicalEventVisitFilterTerm : public CInterfaceOf<ILogicalEventVisitFilterTerm>
```

Comment:
This would almost certainly be simpler with a single class handling and and or with a flag.  The virtual taking a std::function is really not very efficient.  I am not sure if you really need either of these.
(For discission.)